### ISSUE TITLE
terra-clinical-onset-picker lacks validation visuals

### ISSUE THREAD

**Issue by JB016205** [cerner/terra-clinical#779]
# Feature Request

## Description
The current implementation of terra-clinical-onset-picker prevents users from selecting an invalid date, but the user can still manually type in an invalid date. Invalid can come in two forms that we have noticed:
* Invalid format (including locale considerations)
* Before the supplied birthdate

In either case, there is no "isValid" check built into the component to highlight the field red and allow consumers to validate against this for a form check before submission. The end result is _the previous value_ is returned for the date (which could be null) and therefore users could think they submitted a date but it did not submit or submitted a different date than they expected.

Consumers can add this logic at the submit point, but it would be a better experience overall to add visuals around validation and an isValid check to be consistent with other components. There is some semblance of this issue that could also be seen as a defect since if you had a date, and then input an invalid date, the original date is still "submitted". 

## Additional Context / Screenshots
More context can be provided on request.

In the below, the lack of a leading 0 makes this date "invalid" per the component but nothing is clear to the user that this is the case. The same thing occurs if a date is input that is prior to the supplied birth date.
![Screen Shot 2021-10-27 at 9 43 34 AM](https://user-images.githubusercontent.com/3382775/139089138-1de75e37-83d1-4e93-b5bd-4ef03cd1d9e4.png)

## @ Mentions
@ryanthemanuel @benbcai

**Comment by benbcai** [comment]
An internal issue has been logged to track and fix this. This issue will be closed.
